### PR TITLE
Fix arena avatar state sync bug

### PR DESCRIPTION
**Pull request by tyrosine1153**
### Description

- Fix arena avatar state sync bug

### How to test

1. After equipping the equipments or costumes in the arena preperaion, play Arena Battle.
2. Register equipped equipments or costumes in the shop.
3. Check the registered equipments or costumes are unquipped in the arena preperation.

### Related Links
[[아레나] 아레나 전투 준비에 등록된 장비를 상점에 등록할 경우 장착 해제가 되지 않고 출력되는 현상](https://app.asana.com/0/1133453747809944/1202610071912915/f)

### Screenshot
1. After equipping the equipments or costumes in the arena preperaion, play Arena Battle.
    image

2. Register equipped equipments or costumes in the shop.
    image

3. Check the registered equipments or costumes are unquipped in the arena preperation.
    image

### CODE REVIEW

File: nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs [hunk 1]
```diff
@@ -219,6 +204,52 @@ public override void Close(bool ignoreCloseAnimation = false)
 
         #endregion
 
+        private void UpdateAvatarState()
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs:207]:
```suggestion
        private void UpdateArenaAvatarState()
```

File: nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs [hunk 2]
```diff
@@ -219,6 +204,52 @@ public override void Close(bool ignoreCloseAnimation = false)
...
+        {
+            var avatarState = Game.Game.instance.States.CurrentAvatarState;
+            var arenaAvatarState = RxProps.PlayersArenaParticipant.Value.AvatarState;
+            var currentBlockIndex = Game.Game.instance.Agent.BlockIndex;
+
+            var arenaItems = arenaAvatarState.inventory.Items;
+            for (int i = arenaItems.Count - 1; i > 0; i--)
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs:214]:
```
for (int i = arenaItems.Count - 1; i > 0; i--)
```
This `for` loop seems not intuitive. Did you mean it?

File: nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs [hunk 3]
```diff
@@ -219,6 +204,52 @@ public override void Close(bool ignoreCloseAnimation = false)
...
+            {
+                Nekoyume.Model.Item.Inventory.Item existItem;
+                switch (arenaItems[i].item)
+                {
+                    case Equipment arenaEquipment:
+                    {
+                        existItem = avatarState.inventory.Items.FirstOrDefault(item =>
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs:221]: `avatarState.inventory.Items` can change to `avatarState.inventory.Equipments` ?

File: nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs [hunk 4]
```diff
@@ -219,6 +204,52 @@ public override void Close(bool ignoreCloseAnimation = false)
...
+                            item.item is Equipment equipment
+                            && equipment.ItemId == arenaEquipment.ItemId);
+                        break;
+                    }
+                    case Costume arenaCostume:
+                    {
+                        existItem = avatarState.inventory.Items.FirstOrDefault(item =>
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs:228]: `avatarState.inventory.Items` can change to `avatarState.inventory.Costumes` ?

File: nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs [hunk 5]
```diff
@@ -405,7 +436,7 @@ private void Unequip(EquipmentSlot slot, bool considerInventoryOnly)
             }
 
             if (!inventory.TryGetModel(slot.Item, out var targetInventoryItem) ||
-                !(targetInventoryItem.ItemBase is IEquippableItem equippableItem))
+                targetInventoryItem.ItemBase is not IEquippableItem equippableItem)
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs:439]:
```cs
targetInventoryItem.ItemBase is not IEquippableItem equippableItem)
```


Can use `is not` statement in Unity 2021.3.5?

File: nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs (hunk 2 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs:214]: There is a possibility the element of `arenaAvatarState.inventory` will be modified in the loop.
```
line.237
if (existItem is null)
{
    arenaAvatarState.inventory.RemoveItem(arenaItems[i]);
}
```

File: nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs (hunk 3 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/ArenaBattlePreparation.cs:221]: To check `Locked` value later. type of `existItem` must be Item, not itemBase.

**Pull request merged**